### ISSUE TITLE
limit task or job parallelism within a given pipeline

### ISSUE THREAD

**Issue by tkellen** [concourse/concourse#4953]
## What challenge are you facing?
I'm operating a monorepo that self generates pipelines on a per-branch basis. It doesn't happen often, but sometimes there are 20+ projects in a given pipeline, each with their own build/deploy job. Most of the time this workflow is fine (developer bringing up the entire fleet of services dependent on the one they are devving) because we skip builds that have already been performed. However, changes to shared libraries can trigger a build for 20+ java projects that all want to run in parallel. This crushes our CI worker nodes.

## What would make this better?
Having a way to limit the number of overall jobs or tasks running in parallel for a given pipeline or for Concourse as a whole, would be helpful.

Basically this is the exact same issue as https://github.com/concourse/concourse/issues/2628 but at a different level.

## Are you interested in implementing this yourself?
Yes, though I am curious, would it be preferable to the maintainers to implement this as a concourse-wide environment variable or a pipeline-level configuration? Or both?

**Comment by kcmannem** [comment]
hey, we have an experimental feature that gives you the behavior your asking for. It's not of pipeline granularity though, its a global step queue which is customizable to how big your workers are.

https://concourse-ci.org/container-placement.html#limit-active-tasks-strategy

It works well enough for many people and should be an interim fix.

**Comment by tkellen** [comment]
Sorry for the delayed reply. Thank you for the references, I think this will get us through for now!
